File: baekjoon/gg.js
// const fs = require("fs");
// const [a, b] = fs.readFileSync("/dev/stdin").toString().split("\n");

const a = "🔢📟";
const b = "🔢📟📟📟📟";

const emojiMap = {
  "📟": "40404",
  "🔢": "1234",
  "🎰": "777",
  "": "109",
  "💯": "100",
  "🏪": "24",
  "🧑‍🏫": "22",
  "🧑🏻‍🏫": "22",
  "🧑🏼‍🏫": "22",
  "🧑🏽‍🏫": "22",
  "🧑🏾‍🏫": "22",
  "🧑🏿‍🏫": "22",
  "👨‍🏫": "22",
  "👨🏻‍🏫": "22",
  "👨🏼‍🏫": "22",
  "👨🏽‍🏫": "22",
  "👨🏾‍🏫": "22",
  "👨🏿‍🏫": "22",
  "👩‍🏫": "22",
  "👩🏻‍🏫": "22",
  "👩🏼‍🏫": "22",
  "👩🏽‍🏫": "22",
  "👩🏾‍🏫": "22",
  "👩🏿‍🏫": "22",
  "🔞": "18",
  "📅": "17",
  "📆": "17",
  "🔟": "10",
  "9️⃣": "9",
  "🎱": "8",
  "8️⃣": "8",
  "7️⃣": "7",
  "6️⃣": "6",
  "5️⃣": "5",
  "4️⃣": "4",
  "🥉": "3",
  "🏎️": "3",
  "3️⃣": "3",
  "🥈": "2",
  "2️⃣": "2",
  "🥇": "1",
  "🔂": "1",
  "1️⃣": "1",
  "0️⃣": "0",
};

const convertEmoji = (num) => {
  let str = `${num}`;
  const vArr = Object.values(emojiMap);

  for (let i = 0; i < vArr.length; i++) {
    str = str.replaceAll(vArr[i], Object.keys(emojiMap)[i]);
  }

  return str;
};

let aNum = a.trim();
let bNum = b.trim();

for (let i = 0; i < Object.entries(emojiMap).length; i++) {
  aNum = aNum.replaceAll(Object.keys(emojiMap)[i], Object.values(emojiMap)[i]);
  bNum = bNum.replaceAll(Object.keys(emojiMap)[i], Object.values(emojiMap)[i]);
}

const sum = BigInt(aNum) + BigInt(bNum);

console.log(BigInt(aNum), BigInt(bNum));

console.log(sum);

console.log(convertEmoji(sum));

File: baekjoon/gg.go
package main

import (
	"bufio"
	"fmt"
	"math/big"
	"os"
	"strings"
)

func main() {
	num := []string{
		"40404", "1234", "777", "109", "100",
		"24", "22", "22", "22", "22",
		"22", "22", "22", "22", "22",
		"22", "22", "22", "22", "22",
		"22", "22", "22", "22", "18",
		"17", "17", "10", "9", "8",
		"8", "7", "6", "5", "4",
		"3", "3", "3", "2", "2",
		"1", "1", "1", "0"}

	emoji := []string{
		"📟", "🔢", "🎰", "", "💯",
		"🏪", "🧑‍🏫", "🧑🏻‍🏫", "🧑🏼‍🏫", "🧑🏽‍🏫",
		"🧑🏾‍🏫", "🧑🏿‍🏫", "👨‍🏫", "👨🏻‍🏫", "👨🏼‍🏫",
		"👨🏽‍🏫", "👨🏾‍🏫", "👨🏿‍🏫", "👩‍🏫", "👩🏻‍🏫",
		"👩🏼‍🏫", "👩🏽‍🏫", "👩🏾‍🏫", "👩🏿‍🏫", "🔞",
		"📅", "📆", "🔟", "9️⃣", "🎱",
		"8️⃣", "7️⃣", "6️⃣", "5️⃣", "4️⃣",
		"🥉", "🏎️", "3️⃣", "🥈", "2️⃣",
		"🥇", "🔂", "1️⃣", "0️⃣",
	}

	var reader *bufio.Reader = bufio.NewReader(os.Stdin)
	var writer *bufio.Writer = bufio.NewWriter(os.Stdout)

	defer writer.Flush()

	var a, b string
	fmt.Fscanln(reader, &a)
	fmt.Fscanln(reader, &b)

	for i := 0; i < len(num); i++ {
		a = strings.ReplaceAll(a, emoji[i], num[i])
		b = strings.ReplaceAll(b, emoji[i], num[i])
	}

	aNum := new(big.Int)
	bNum := new(big.Int)

	aNum, _ = aNum.SetString(a, 10)
	bNum, _ = bNum.SetString(b, 10)

	sum := new(big.Int)

	sumString := sum.Add(aNum, bNum).String()

	for i := 0; i < len(num); i++ {
		sumString = strings.ReplaceAll(sumString, num[i], emoji[i])
	}

	writer.WriteString(sumString)
}
